### DISCUSSION TITLE
Postgres db version upgrade supported?

### DISCUSSION THREAD

**Discussion by galaczi** [coollabsio/coolify#3474] [Q&A]
Hi, I have just deployed my first Coolify and loving it. Just wondering for future reference, does it handle db version upgrade? From past experience it's not only a matter of updating docker version with the same volume. The volume data should be updated too.

**Comment by notflip** [comment]
Diud you ever find the solution @galaczi ? I want to bump my postgres version

**Reply by galaczi** [reply]
I did not unfortunately.

**Reply by notflip** [reply]
@galaczi (not recommended!) I changed the name of my image from postgres:16-alpine to postgres:17.4-alpine, and then restarted the container, this worked, but my data is corrupted because of the upgrade

**Reply by djsisson** [reply]
you need to run pg_upgrade if you are doing that.

https://www.postgresql.org/docs/current/pgupgrade.html

alternatively, start a new db project with the new version, then backup from your old db and restore into your new db

**Comment by KianAnbarestani** [comment]
@galaczi 
Coolify doesn’t handle internal DB migrations during version upgrades—it only updates the Docker image. For major version changes, you should back up your data and manually migrate it using the DB's official tools. This ensures data integrity and avoids volume corruption. Always test upgrades on a staging copy first.
